app/facades/pro.rb

```diff
@@ -0,0 +1,70 @@
+class Pro < Dashboard
+  def initialize(user)
+    super
+  end
+
+  def user_article_ids
+    user.articles.pluck(:id)
+  end
+
+  def this_week_reactions
+    ChartDecorator.decorate(Reaction.where(reactable_id: user_article_ids, reactable_type: "Article").where("created_at > ?", 1.week.ago).order("created_at ASC"))
+  end
+
+  def this_week_reactions_count
+    this_week_reactions.size
+  end
+
+  def last_week_reactions_count
+    Reaction.where(reactable_id: user_article_ids, reactable_type: "Article").where("created_at > ? AND created_at < ?", 2.weeks.ago, 1.week.ago).size
```

Comment:
It's tempting to create private methods like `reactions_scope` and `comments_scope` for the repeating parts, just a suggestion.

Reply:
Yeah, I think I'll do that. There's definitely a better way to organize the mass amount of data.